src/gui/src/UI/UIDesktop.js

```diff
@@ -1385,6 +1440,22 @@ $(document).on('click', '.qr-btn', async function (e) {
     });
 })
 
+$(document).on('click', '.notifications-history-btn', function(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    
+    // Toggle active state of the button
+    $(this).toggleClass('active');
+    
+    // Create or show notifications sidebar
+    UIWindowNotifications();
+})
+
+$(document).on('click', '.notification', function() {
+    // Update badge after notification is clicked
+    setTimeout(window.update_notification_badge_count, 100);
```

Comment:
Why are we polling so frequently here? If I merged this as-is, I wouldn't be surprised if it had a noticeable impact on `puter.com`. We get notifications over websockets, so we can usually update the count there; this should then be a very infrequent poll (several minutes) just in case the badge goes out of sync.